### PR TITLE
Fix visibility of PhantomData in ABI types

### PR DESCRIPTION
**Pull request by zec**
I ran into this issue when trying (unsuccessfully) to implement COM interfaces with generics on custom types. As things stand in `master`, everything is `pub` in the generated the `*_abi` structs... except for the `PhantomData<_>` used to mark the generic parameters. This fixes that.

This doesn't do much right now, but should be helpful once generics support is plumbed through `#[implement]`.

### CODE REVIEW

**Comment by zec** [comment]
I have a PoC of this enabling the ability to implement interfaces with generics at https://github.com/zec/windows-rs-generic-test.

File: crates/gen/src/types/delegate.rs [hunk 1]
```diff
@@ -36,7 +36,9 @@ impl Delegate {
         let abi_signature = signature.gen_winrt_abi(gen);
         let fn_constraint = signature.gen_winrt_constraint(gen);
         let guid = self.0.gen_guid(gen);
-        let phantoms = self.0.gen_phantoms(gen);
+        let intf_struct_phantoms = self.0.gen_phantoms(gen);
```
Comment (kennykerr) [inline crates/gen/src/types/delegate.rs:39]: I'd prefer to call this simply `struct_phantoms` - that's sufficiently unambiguous.

File: crates/gen/src/types/interface.rs [hunk 2]
```diff
@@ -81,7 +81,8 @@ impl Interface {
         let name = self.0.gen_name(gen);
         let guid = self.0.gen_guid(gen);
         let abi_name = self.0.gen_abi_name(gen);
-        let phantoms = self.0.gen_phantoms(gen);
+        let intf_phantoms = self.0.gen_phantoms(gen);
```
Comment (kennykerr) [inline crates/gen/src/types/interface.rs:84]: Same here: `struct_phantoms`.

**Review by kennykerr — CHANGES_REQUESTED** [review]
Thanks, this looks great! Now that 0.4.0 is published, I'm planning to spend most of my time working on authoring support for WinRT and COM interfaces. 

Just a few minor tweaks to the naming.

File: crates/gen/src/types/delegate.rs [hunk 3]
```diff
@@ -81,7 +83,7 @@ impl Delegate {
         quote! {
             #[repr(transparent)]
             #[derive(::std::cmp::PartialEq, ::std::cmp::Eq, ::std::clone::Clone, ::std::fmt::Debug)]
-            pub struct #name(::windows::IUnknown, #phantoms) where #constraints;
+            pub struct #name(::windows::IUnknown, #(#intf_struct_phantoms,)*) where #constraints;
```
Comment (kennykerr) [inline crates/gen/src/types/delegate.rs:86]:
```suggestion
            pub struct #name(::windows::IUnknown, #(#struct_phantoms,)*) where #constraints;
```

File: crates/gen/src/types/interface.rs [hunk 4]
```diff
@@ -149,7 +150,7 @@ impl Interface {
             #[repr(transparent)]
             #[derive(::std::cmp::PartialEq, ::std::cmp::Eq, ::std::clone::Clone, ::std::fmt::Debug)]
             #hidden
-            pub struct #name(::windows::Object, #phantoms) where #constraints;
+            pub struct #name(::windows::Object, #(#intf_phantoms,)*) where #constraints;
```
Comment (kennykerr) [inline crates/gen/src/types/interface.rs:153]:
```suggestion
            pub struct #name(::windows::Object, #(#struct_phantoms,)*) where #constraints;
```

**Comment by zec** [comment]
OK, I've applied the naming changes you suggested.

**Review by kennykerr — APPROVED** [review]

**Pull request merged**